internal/js/modules/k6/browser/common/frame.go

```diff
@@ -1915,6 +1915,10 @@ func (f *Frame) WaitForNavigation(opts *FrameWaitForNavigationOptions) (*Respons
 	select {
 	case evt := <-navEvtCh:
 		if e, ok := evt.(*NavigationEvent); ok {
+			if e.err != nil {
+				return nil, e.err
```

Comment:
Here's the key to surface the errors, previously we were ignoring if the `NavigationEvent` was holding an `error`.